### DISCUSSION TITLE
Accessing data in packages after loading model

### DISCUSSION THREAD

**Discussion by wkitlasten** [modflowpy/flopy#1442] [General]
I am struggling to access and manipulate data from a gwf model. In this specific case the GHB data is stored as an external file. I  load the model in flopy:

gwf = sim.get_model(model)

But I am having difficulting figuring out how to access the data. An example of what I tried is:

```
gwf.ghb.data_list
Out[23]: 
[,
 ,
 ,
 ,
 ,
 ,
 ,
 ,
 ,
 {internal}
 (11464),
 ]
```

Does the {internal} suggest the data is not set to external? Because the .ghb file suggests otherwise:

```
# File generated by Flopy version 3.3.5 on 07/08/2022 at 13:33:20.
BEGIN options
END options

BEGIN dimensions
  MAXBOUND  11464
END dimensions

BEGIN period  1
    OPEN/CLOSE  'w_1000_10m.ghb_coast.dat'
END period  1
```

Is there a method to access the data?

Alternatively, is there a method that returns the external filename, which I can load myself (e.g., in pandas)?

Thanks

**Comment by christianlangevin** [comment]
gwf.ghb.stress_period_data.get_data() should return the information in a way you can work with it.  @spaulins-usgs, I don't see this described very well in the docs, especially how to work with stress period data in external files.  Could we improve the docs a bit?  I think the documentation for gridded data, like hydraulic conductivity, is much better.

**Comment by wkitlasten** [comment]
That does it. Thanks. (and thanks for moving to the "Discussions")

**Comment by spaulins-usgs** [comment]
@langevin-usgs, I will update tutorial 8 with examples of storing external stress period list data.

In addition to that, I think we can also improve the flopy code docs so that they document the attributes of each package.  For example, the documentation for the ghb package class

https://flopy.readthedocs.io/en/3.3.5/_notebooks/tutorial06_mf6_data.html#Adding-Stress-Period-List-Data

defines all of the constructor parameters including stress_period_data, but does not define the class attributes.  The stress_period_data attribute should be defined here as MFDataList (stress_period_data is both a parameter and an attribute of the ModflowGwfghb class, where the parameter is a different type than the attribute).

I will work on documenting the class attributes for all package data.  While the MF2005 docs also do not document package attributes, for now I will focus on updating the MF6 docs.  @langevin-usgs, let me know if you want to discuss this further.
